go/internal/types/uuid.go

```diff
@@ -0,0 +1,198 @@
+package types
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/goccy/go-json"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/google/uuid"
+)
+
+type UUIDBuilder struct {
+	*array.ExtensionBuilder
+}
```

Comment:
instead of this being duplicated now in `arrow/internal/testing/types/` and here, could we remove it from `arrow/internal/testing/types/` and anywhere that references `UUIDType` in the arrow pkg also points to this spot? That way both are sharing this implementation